src/arraydist.jl

```diff
@@ -1,3 +1,10 @@
+"""
+    arraydist(dists)
+
+Create a distribution from an array of distributions.
+"""
+arraydist(dists::AbstractArray) = product_distribution(dists)
```

Comment:
Should we limit this fallback to
```suggestion
arraydist(dists::AbstractArray{<:Distribution}) = product_distribution(dists)
```
?